functional-tests/tests/bridge_withdraw_happy_op_return.py

```diff
@@ -0,0 +1,127 @@
+        el_address = self.eth_account.address
+        payload = "hello world"
+        bosd = string_to_opreturn_descriptor(payload)
+        self.debug(f"BOSD: {bosd}")
+
+        self.debug(f"Address: {address}")
+        self.debug(f"Change Address: {withdraw_address}")
+        self.debug(f"EL Address: {el_address}")
+
+        # Original BTC balance
+        btc_url = self.btcrpc.base_url
+        btc_user = self.btc.get_prop("rpc_user")
+        btc_password = self.btc.get_prop("rpc_password")
+        original_balance = get_balance(withdraw_address, btc_url, btc_user, btc_password)
+        self.debug(f"BTC balance before withdraw: {original_balance}")
+
+        # Make sure starting ETH balance is 0
+        check_initial_eth_balance(self.rethrpc, el_address, self.debug)
+
+        bridge_pk = get_bridge_pubkey(self.seqrpc)
+        self.debug(f"Bridge pubkey: {bridge_pk}")
+
+        # make two deposits
+        self.deposit(ctx, el_address, bridge_pk)
+        self.deposit(ctx, el_address, bridge_pk)
+
+        # Withdraw
+        l2_tx_hash, _tx_receipt, _total_gas_used = self.withdraw_op_return(ctx, el_address, bosd)
+        l2_block_hash = rethrpc.eth_getTransactionByHash(l2_tx_hash)["blockHash"]
+
+        # Make sure that the OP_RETURN is in the Strata block execution update
+        # FIXME: I have no idea how to do this
+        exec_update = seqrpc.strata_getExecUpdateById(l2_block_hash)
+        self.debug(f"Exec update: {exec_update}")
+        print(f"Exec update: {exec_update}")  # FIXME: REMOVE ME
+        update_idx = exec_update["update_idx"]
+        self.debug(f"Exec update index: {update_idx}")
+        print(f"Exec update index: {update_idx}")  # FIXME: REMOVE ME
+        sync_event = seqrpc.strata_getSyncEvent(update_idx)
+        self.debug(f"Sync event: {sync_event}")
+        print(f"Sync event: {sync_event}")  # FIXME: REMOVE ME
+
+        withdrawals = exec_update["withdrawals"]
+        self.debug(f"Exec update withdrawals: {withdrawals}")
+        print(f"Exec update withdrawals: {withdrawals}")  # FIXME: REMOVE ME
+
+        # Move forward a single block
+        block = generate_n_blocks(btcrpc, 1)[0]  # There's only one
+        last_block = btcrpc.getblock(block)
+
+        # Get all withdraw bridge messages
+        # VODepositSig(10)
+        scope = "00"
+        self.debug(scope)
+
+        msgs = seqrpc.strata_getBridgeMsgsByScope(scope)
+        self.debug(msgs)
+        print(f"Bridge msgs: {msgs}")  # FIXME: REMOVE ME
+
+        # Get the output of the tx
+        # OP_RETURN is the second output
+        # FIXME: I have no idea how to do this
+        outputs = last_block["txs"][0].as_dict()["outputs"]
+        op_return_output = outputs[1]
+        self.debug(f"OP_RETURN output: {op_return_output}")
+        op_return_script_type = op_return_output["script_type"]
+        assert op_return_script_type == "nulldata", "OP_RETURN not found"
+
+        return True
```

Comment:
Any ideas on how to test this guy?

Reply:
Ah we should have an RPC that fetches the pending withdrawal intents and withdrawal assignments.  That's probably the *right way* to do this.  Do you want to do this here or create a new PR for it and rebase?

Reply:
To be honest, we can disable this test and open a new ticket to add the RPC and fix the withdraw OP_RETURN test.
What do you think?
This is touching a LOT of the codebase. I much prefer to defer the functional test to later than have this in a rebase purgatory.

Reply:
https://alpenlabs.atlassian.net/browse/STR-919